core/corehttp/gateway_handler.go

```diff
@@ -338,10 +338,8 @@ func (i *gatewayHandler) putHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	tctx, cancel := context.WithTimeout(ctx, time.Minute)
-	defer cancel()
```

Comment:
yeah, we should keep this one. anything on the gateway should have an explicit timeout (although, maybe it should be made configurable)